grails-app/controllers/payment/ContactsController.groovy

```diff
@@ -136,103 +136,107 @@ class ContactsController {
         command.gender = contact.gender
         command.surveyVoteWeight = contact.surveyVoteWeight
         BasicDataKuorumUserRSDTO contactUser = kuorumUserService.findBasicUserRSDTO(contact.mongoId, true)
-        [command:command,contact:contact,contactUser:contactUser]
+        [command: command, contact: contact, contactUser: contactUser]
     }
 
-    def updateContact(ContactCommand command){
+    def updateContact(ContactCommand command) {
         KuorumUserSession user = springSecurityService.principal
         ContactRSDTO contact = contactService.getContact(user, command.contactId)
-        if (command.hasErrors() || !contact){
+        if (command.hasErrors() || !contact) {
             BasicDataKuorumUserRSDTO contactUser = kuorumUserService.findBasicUserRSDTO(contact.mongoId, true)
-            render view: "/contacts/editContact", model:[command:command,contact:contact,contactUser:contactUser]
+            render view: "/contacts/editContact", model: [command: command, contact: contact, contactUser: contactUser]
             return
         }
         contact.name = command.name
         contact.email = command.email
         // If the fields are null will not be updated on API.
-        contact.phonePrefix = command.phonePrefix==null?"":command.phonePrefix
-        contact.phone = command.phone==null?"":command.phone
-        contact.surname = command.surname==null?"":command.surname
-        contact.externalId = command.externalId==null?"":command.externalId
+        contact.phonePrefix = command.phonePrefix == null ? "" : command.phonePrefix
+        contact.phone = command.phone == null ? "" : command.phone
+        contact.surname = command.surname == null ? "" : command.surname
+        contact.externalId = command.externalId == null ? "" : command.externalId
         contact.language = command.language
         contact.birthdate = command.birthDate
         contact.gender = command.gender
         contact.surveyVoteWeight = command.surveyVoteWeight
-        ContactRSDTO contactUpdated = contactService.updateContact(user, contact, contact.getId())
-        flash.message=g.message(code: 'tools.contact.edit.success', args: [contact.name])
-        redirect(mapping:"politicianContactEdit", params: [contactId: contactUpdated.getId()])
+        try {
+            ContactRSDTO contactUpdated = contactService.updateContact(user, contact, contact.getId())
+            flash.message = g.message(code: 'tools.contact.edit.success', args: [contact.name])
+        } catch (Exception e) {
+            flash.error = g.message(code: 'tools.contact.edit.phonenumber.error')
```

Comment:
You are catching a generic exception. Change the key to 'tools.contact.edit.error' and put a generic error. If you want to have a fined handler, e has the Exception thrown by the api wrapped inself.